libs/domains/services/feature/src/lib/service-action-toolbar/service-action-toolbar.tsx

```diff
@@ -0,0 +1,199 @@
+import { isDeleteAvailable, isDeployAvailable, isRedeployAvailable, isStopAvailable } from '@qovery/shared/util-js'
+import { useDeleteService } from '../hooks/use-delete-service/use-delete-service'
+import { useDeployService } from '../hooks/use-deploy-service/use-deploy-service'
+import { useDeploymentStatus } from '../hooks/use-deployment-status/use-deployment-status'
+import { useRedeployService } from '../hooks/use-redeploy-service/use-redeploy-service'
+import { useService } from '../hooks/use-service/use-service'
+import { useStopService } from '../hooks/use-stop-service/use-stop-service'
+
+export function ServiceActionToolbar({ serviceId }: { serviceId: string }) {
+  const { organizationId = '', projectId = '', environmentId = '' } = useParams()
+  const navigate = useNavigate()
+  const { openModalConfirmation } = useModalConfirmation()
+  const { data: environment } = useEnvironment({ environmentId })
+  const { data: service } = useService({ environmentId, serviceId })
+  const { data: deploymentStatus } = useDeploymentStatus({ environmentId, serviceId })
+
+  const { mutate: deployService } = useDeployService({ environmentId })
+  const { mutate: redeployService } = useRedeployService({ environmentId })
+  const { mutate: stopService } = useStopService({ environmentId })
+  const { mutateAsync: deleteService } = useDeleteService({ environmentId })
+
+  const [, copyToClipboard] = useCopyToClipboard()
+  const copyContent = `Organization ID: ${organizationId}\nProject ID: ${projectId}\nEnvironment ID: ${environmentId}\nService ID: ${serviceId}`
+
+  if (!service || !deploymentStatus) return <Skeleton height={32} width={115} />
+
+  const mutationDelete = async () => {
+    openModalConfirmation({
+      title: 'Delete service',
+      name: service.name,
+      isDelete: true,
+      action: async () => {
+        try {
+          await deleteService({ serviceId, serviceType: service.serviceType })
+          navigate(SERVICES_URL(organizationId, projectId, environmentId) + SERVICES_GENERAL_URL)
+        } catch (error) {
+          console.error(error)
+        }
+      },
+    })
+  }
+
+  const mutationDeploy = () => deployService({ serviceId, serviceType: service.serviceType })
+
+  const mutationRedeploy = async () => {
+    openModalConfirmation({
+      mode: environment?.mode,
+      title: 'Confirm redeploy',
+      description: 'To confirm the redeploy of your service, please type the name:',
+      name: service.name,
+      action: () => redeployService({ serviceId, serviceType: service.serviceType }),
+    })
+  }
+
+  const mutationStop = async () => {
+    openModalConfirmation({
+      mode: environment?.mode,
+      title: 'Confirm stop',
+      description: 'To confirm the stopping of your service, please type the name:',
+      name: service.name,
+      action: () => stopService({ serviceId, serviceType: service.serviceType }),
+    })
+  }
+
+  const menuManageDeployment = (
```

Comment:
optional: you can probably extract that into its own `function MenuManageDeployment` to split hooks logic but also avoid recreation of the whole tree at every render